deploy using tiup: align three PRs (#8603)
pingcap/docs: production-deployment-using-tiup.md

source /home/tidb/.bash_profile
```

The `local_install.sh` script automatically runs the `tiup mirror set tidb-community-server-${version}-linux-amd64` command to set the current mirror address to `tidb-community-server-${version}-linux-amd64`.

To switch the mirror to another directory, run the `tiup mirror set <mirror-dir>` command. To switch the mirror to the online environment, run the `tiup mirror set https://tiup-mirrors.pingcap.com` command.

</div>
</SimpleTab>

## Step 3. Initialize cluster topology file

Run the following command to create a cluster topology file:

{{< copyable "shell-regular" >}}

```shell
tiup cluster template > topology.yaml
```

In the following two common scenarios, you can generate recommended topology templates by running commands:

> - Hybrid deployment: Multiple instances are deployed on a single machine. For details, see [Hybrid Deployment Topology](/hybrid-deployment-topology.md).

Reviewer: ```suggestion
- For hybrid deployment: Multiple instances are deployed on a single machine. For details, see [Hybrid Deployment Topology](/hybrid-deployment-topology.md).
```

    {{< copyable "shell-regular" >}}

    ```shell
    tiup cluster template --full > topology.yaml
    ```

> - Geo-distributed deployment: TiDB clusters are deployed in geographically distributed data centers. Fordetails, see [Geo-Distributed Deployment Topology](/geo-distributed-deployment-topology.md).

Reviewer: ```suggestion
- For geo-distributed deployment: TiDB clusters are deployed in geographically distributed data centers. For details, see [Geo-Distributed Deployment Topology](/geo-distributed-deployment-topology.md).
```

    {{< copyable "shell-regular" >}}

    ```shell
    tiup cluster template --multi-dc > topology.yaml
    ```

> Execute `vi topology.yaml` to see the configuration file content:

Reviewer: ```suggestion
Run `vi topology.yaml` to see the configuration file content:
```

  - host: 10.0.1.4
```

The following examples cover seven common scenarios. You need to modify the configuration file (named `topology.yaml`) according to the topology description and templates in the corresponding links. For other scenarios, edit the configuration template accordingly.

| Application | Configuration task | Configuration file template | Topology description |
| :-- | :-- | :-- | :-- |

> | OLTP | [Deploy minimal topology](/minimal-deployment-topology.md) | <ul><li>[Simple minimal configuration template](/config-templates/simple-mini.yaml) </li> <li> [Full minimal configuration template](/config-templates/complex-mini.yaml) | This is the basic cluster topology, including tidb-server, tikv-server, and pd-server. |

Reviewer: ```suggestion
| OLTP | [Deploy minimal topology](/minimal-deployment-topology.md) | <ul><li>[Simple minimal configuration template](/config-templates/simple-mini.yaml) </li> <li> [Full minimal configuration template](/config-templates/complex-mini.yaml) </li> </ul>| This is the basic cluster topology, including tidb-server, tikv-server, and pd-server. |
```

> - If password-free login to the target machine has been configured, no authentication is required.
>
> In general, TiUP creates the user and group specified in the `topology.yaml` file on the target machine, with the following exceptions:
>
> - The user name configured in `topology.yaml` already exists on the target machine.
> - You have used the `--skip-create-user` option in the command line to explicitly skip the step of creating the user.

Before you run the `deploy` command, use the `check` and `check --apply` commands to detect and automatically repair potential risks in the cluster:

1. Check for potential risks:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup cluster check ./topology.yaml --user root [-p] [-i /home/root/.ssh/gcp_rsa]
    ```

2. Enable automatic repair:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup cluster check ./topology.yaml --apply --user root [-p] [-i /home/root/.ssh/gcp_rsa]
    ```

3. Deploy a TiDB cluster:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup cluster deploy tidb-test v6.0.0 ./topology.yaml --user root [-p] [-i /home/root/.ssh/gcp_rsa]
    ```

> In the preceding command:

Reviewer: ```suggestion
In the `tiup cluster deploy` command above:
```

- `tidb-test` is the name of the deployed TiDB cluster.

> - `v6.0.0` is the version of the deployed TiDB cluster. You can see the latest supported versions by running `tiup list tidb`.

Reviewer: ```suggestion
- `v6.0.0` is the version of the TiDB cluster to be deployed. You can see the latest supported versions by running `tiup list tidb`.
```
